packages/extension/src/ui/features/accountTokens/tokens.state.ts

```diff
@@ -323,7 +263,40 @@ export const useAllTokens = (
     }
   )
 
-  return userTokens || []
+  const error = useMemo(() => {
+    const errorCode =
+      maybeSuppressError?.status || maybeSuppressError?.errorCode
+    if (!SUPPRESS_ERROR_STATUS.includes(errorCode)) {
+      return maybeSuppressError
+    }
+  }, [maybeSuppressError])
+
+  const tokenDetailsIsInitialising = !error && !userTokens
+
+  // refetch when number of pending transactions goes down
```

Comment:
In our case, it should refetch when there are new transactions in `pendingTransactions, since we update balance when there are new transactions in mempool

Reply:
Thanks, will update

Reply:
@h0ngcha0 This issue seems not resolved yet?

Reply:
I think this issue is resolve [here](https://github.com/alephium/extension-wallet/blob/improve-get-balance/packages/extension/src/ui/features/accountTokens/tokens.state.ts#L277-L281)

Reply:
That fix can be problematic. Imaging there is one mempool tx in the mempool, if I submit a new mempool tx and the old one is removed at the same time, then the number of mempool transaction does not change.

Reply:
That is true, thanks :+1: 

We can also compare the actual pending txs and see if there are actually new txs pending, WDYT?

Reply:
Yep, that would work